### PR TITLE
Take spvgen.h with ability to load spvgen.so from specified directory

### PR DESCRIPTION
**Pull request by trenouf**
Change-Id: I9b05670c1fed7cd351f63cf7f87947d96e9e25e5

### CODE REVIEW

File: imported/spirv/spvgen.h [hunk 1]
```diff
@@ -412,7 +409,8 @@ static const char* SpvGeneratorName = "spvgen.so";
 // =====================================================================================================================
 // Initialize SPIR-V generator entry-points
 // This can be called multiple times in the same application.
-bool InitSpvGen()
+bool InitSpvGen(
+    const char* pSpvGenDir)   // Directory to load SPVGEN library from, or "" to use OS's default search path
```
Comment (amdrexu) [inline imported/spirv/spvgen.h]: // -> // [in]

File: imported/spirv/spvgen.h [hunk 2]
```diff
@@ -421,10 +419,19 @@ bool InitSpvGen()
     }
 
     bool success = true;
+    const char* pLibName = SpvGeneratorName;
```
Comment (amdrexu) [inline imported/spirv/spvgen.h]: pLibName -> pFullFileName

File: imported/spirv/spvgen.h (hunk 1 above)
Comment (trenouf) [reply imported/spirv/spvgen.h]: Now fixed in https://github.com/GPUOpen-Drivers/spvgen/pull/3
I will update this commit a bit later on.

File: imported/spirv/spvgen.h (hunk 2 above)
Comment (trenouf) [reply imported/spirv/spvgen.h]: Disagree with this misleading name. It is the full name only in the case that no directory was provided.

File: imported/spirv/spvgen.h (hunk 1 above)
Comment (trenouf) [reply imported/spirv/spvgen.h]: Fixed here as well now.

File: imported/spirv/spvgen.h [hunk 3]
```diff
@@ -421,10 +419,19 @@ bool InitSpvGen()
...
+    if (pSpvGenDir != nullptr)
+    {
+        libNameBuffer = pSpvGenDir;
+        libNameBuffer += "/";
+        libNameBuffer += pLibName;
+        pLibName = libNameBuffer.c_str();
+    }
 #ifdef _WIN32
-    HMODULE hModule = LoadLibrary(SpvGeneratorName);
+    HMODULE hModule = LoadLibrary(pLibName);
```
Comment (linqun) [inline imported/spirv/spvgen.h]: please replace LoadLibrary with LoadLibraryA, otherwise, build will fail when UNICODE is defined.

Comment (trenouf) [reply imported/spirv/spvgen.h]: Good point. I made that change in a different internal version of the patch, but I forgot to transfer it back to this version.

File: imported/spirv/spvgen.h (hunk 2 above)
Comment (amdrexu) [reply imported/spirv/spvgen.h]: OK. Please ignore this.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 4c71f6bdeced55b0c74f9e16daf7cd8fbec94e13
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 99901922d67e502ff4233a8bfcbfa0347c0ffa1b</li>
<li>SPVGEN: dev 69311f7c1e220d190909298b9c7bd702ecc237ff</li>
<li>XGL: dev 5b4058dc288a25f6554fcc61f80bf3f27eb35d8d</li>
<li>LLVM: amd-vulkan-dev 97cc33415120ae3ed472b6dd5cb234b74a80bd80</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 901/901 (100.0%)</li>
<li>Failed: 0/901 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.1.2-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 312622/312622 (100.0%)</li>
<li>Failed: 0/312622 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Vega20</summary><ul>

<li>Passed: 312687/312687 (100.0%)</li>
<li>Failed: 0/312687 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 12e716f984069124a9321bf047131ce278dae3d1
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 99901922d67e502ff4233a8bfcbfa0347c0ffa1b</li>
<li>SPVGEN: dev 69311f7c1e220d190909298b9c7bd702ecc237ff</li>
<li>XGL: dev 5b4058dc288a25f6554fcc61f80bf3f27eb35d8d</li>
<li>LLVM: amd-vulkan-dev 97cc33415120ae3ed472b6dd5cb234b74a80bd80</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 901/901 (100.0%)</li>
<li>Failed: 0/901 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.1.2-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 312622/312622 (100.0%)</li>
<li>Failed: 0/312622 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Vega20</summary><ul>

<li>Passed: 312687/312687 (100.0%)</li>
<li>Failed: 0/312687 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 942d529b35de0e8870d4265b8eec40c54699a180
<details open><summary>Driver commits used in build</summary><ul>

<li>PAL: dev 99901922d67e502ff4233a8bfcbfa0347c0ffa1b</li>
<li>SPVGEN: dev 69311f7c1e220d190909298b9c7bd702ecc237ff</li>
<li>XGL: dev 5b4058dc288a25f6554fcc61f80bf3f27eb35d8d</li>
<li>LLVM: amd-vulkan-dev 97cc33415120ae3ed472b6dd5cb234b74a80bd80</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 901/901 (100.0%)</li>
<li>Failed: 0/901 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.1.1.2-cherry-picks)</summary><ul>


<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 312622/312622 (100.0%)</li>
<li>Failed: 0/312622 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Vega20</summary><ul>

<li>Passed: 312687/312687 (100.0%)</li>
<li>Failed: 0/312687 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by trenouf** [comment]
Hi Jacob. Can you merge this please so I can rebase the amdllpc -spvgen-dir change on top of it? Thanks.

**Comment by JacobHeAMD** [comment]
Sorry for the delay,Tim.

**Comment by trenouf** [comment]
No problem; thanks.

**Pull request merged**